pac4j-saml/src/test/java/org/pac4j/saml/client/AuthnRequestInflator.java

```diff
@@ -0,0 +1,37 @@
+package org.pac4j.saml.client;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.zip.Inflater;
+import java.util.zip.InflaterInputStream;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+public class AuthnRequestInflator {
+
+    public static String getInflatedAuthnRequest(final String location) {
+        final List<NameValuePair> pairs = URLEncodedUtils.parse(java.net.URI.create(location), "UTF-8");
+        final Inflater inflater = new Inflater(true);
+        final byte[] decodedRequest = Base64.getDecoder().decode(pairs.get(0).getValue());
+        final ByteArrayInputStream is = new ByteArrayInputStream(decodedRequest);
+        final InflaterInputStream inputStream = new InflaterInputStream(is, inflater);
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
+        String line;
+        final StringBuilder bldr = new StringBuilder();
+        try {
+            while ((line = reader.readLine()) != null) {
+                bldr.append(line);
+            }
+        } catch (final IOException e) {
+            throw new RuntimeException(e);
+        }
+        return bldr.toString();
+    }
```

Comment:
You'll need:

1. Javadoc for the method
2. Streams must all be closed
3. This is effectively a utility class with only one static method and should be designed as such.

Reply:
I just moved this existing code to a separate class so it can be reused...
But OK, I'll do the changes.

Reply:
Yes, it's that now we see it ;-) I also like to have Javadoc on the class with the `@since` and `@author` annotations...

Reply:
:-D
Class-level Javadoc pushed.